internal/rdb/rdb.go

```diff
@@ -869,6 +896,9 @@ return redis.status_reply("OK")`)
 // Archive sends the given task to archive, attaching the error message to the task.
 // It also trims the archive by timestamp and set size.
 func (r *RDB) Archive(ctx context.Context, msg *base.TaskMessage, errMsg string) error {
+	if *(r.config.MaxArchiveSize) <= 0 {
+		return nil
```

Comment:
Don't we need `return r.DeleteTask(msg.Queue, msg.ID)` here, to ensure that the task is deleted instead of just left in the original queue?

Reply:
This is fine as the author intended that setting anything < 1 should never delete anything. But, another issue is that if the configs are set < 0, a default value of 1 is used that will trigger an archive which is not intended behavior.